pkg/operator/operator.go

```diff
@@ -811,6 +811,31 @@ func (op *Reporting) newPrometheusConn(promConfig promapi.Config) (prom.API, err
 	return prom.NewAPI(client), nil
 }
 
+func (op *Reporting) handleExpiredReport(report *metering.Report, now time.Time) error {
+	// check if the Report is being deleted already
+	if report.DeletionTimestamp != nil {
+		return fmt.Errorf("report was already marked for deletion")
+	}
+	// check if the Report is past its retention time
+	if reportExpired := isReportExpired(op.logger, report, now); !reportExpired {
+		op.logger.Debugf("the %s Report in the %s namespace has not reached the expiration date", report.Name, report.Namespace)
+		return nil
+	}
+	// check if the Report is used by any other Report or ReportQuery, if not delete it
+	if reportIsNotInput := isReportNotUsedAsInput(report, op); reportIsNotInput {
+		op.deleteReport(report)
+		newDeletionTimestamp := metav1.Now()
+		report.SetDeletionTimestamp(&newDeletionTimestamp)
```

Comment:
Hmm, this seems like something that would get us in trouble. We shouldn't explicitly call `op.deleteReport(report)` as that event will get picked up when we make the client-go delete call anyways.

Reply:
After some more looking and thinking, agreed.